lib/main.dart

```diff
@@ -62,14 +71,40 @@ class _LifecycleAwareAppState extends State<LifecycleAwareApp>
   Widget build(BuildContext context) {
     return GetMaterialApp(
       debugShowCheckedModeBanner: false,
-      initialRoute: Get.find<AuthController>().user == null ? '/' : 'dashboard',
+      initialRoute: determineStartingPage(),
       theme: appThemeData,
       defaultTransition: Transition.fade,
       getPages: AppPages.pages,
     );
   }
 }
 
+Future<void> setupCurrentUser() async {
+  final FirebaseUser currentUser = await FirebaseAuth.instance.currentUser();
+  if(currentUser != null) {
+    final User user = User.fromFirebase(currentUser, LoginType.google);
+    Get.find<AuthController>().setUser(user);
+    final UserDataController _userDataController = Get.put(UserDataController(UserDataRepository(), user));
+    await _userDataController.loadAuxiliaryInfoForUser();
+  }
+}
+
+String determineStartingPage() {
+  if(Get.find<AuthController>().user == null) {
+    return '/';
+  }
+  else {
+    print(Get.find<UserDataController>().phoneNumberAssociated);
```

Comment:
is the `UserDataController` guaranteed to be in the global when you reach this point?

Reply:
Yes, there are 2 ways for the `UserDataController` to be initialized.
It is either in the global when we detect currentUser is not null in `setupCurrentUser` in main
or it is created when we sign in with `AuthController.signInWithGoogle()`

Reply:
If the user was not null (which is what is required to reach this point), it would have been created previously in `setupCurrentUser`.

Reply:
Ahh I see, okay, thanks for the clarification! I think it would be better to put some comments about this to help people who read the code in the future. Or perhaps could also do some assertions or error handling in case the `find()` operation returns null

Reply:
Yeah I will add some comments

Reply:
Added more comments :)